web/src/main/java/org/mskcc/cbio/oncokb/api/pvt/PrivateUtilsApiController.java

```diff
@@ -142,4 +148,73 @@
+        List<MatchVariantResult> results = new ArrayList<>();
+
+        for (Query query : body.getQueries()) {
+            MatchVariantResult matchVariantResult = new MatchVariantResult();
+            matchVariantResult.setQuery(query);
+            Set<MatchVariant> match = new HashSet<>();
+
+            for (MatchVariant matchVariantRequestVariant : body.getOncokbVariants()) {
+                if(query.getHugoSymbol().equals(matchVariantRequestVariant.getHugoSymbol())) {
+                    boolean isMatch = matchVariant(query.getHugoSymbol(), matchVariantRequestVariant.getAlteration(), query.getAlteration());
+                    if (isMatch) {
+                        match.add(matchVariantRequestVariant);
+                    }
+                }
+            }
+            matchVariantResult.setResult(match);
+            results.add(matchVariantResult);
+        }
+        return new ResponseEntity<>(results, HttpStatus.OK);
+    }
+
+    @Override
+    public ResponseEntity<Map<String, Boolean>> validateVariantExampleGet(String hugoSymbol, String variant, String examples) throws ParserConfigurationException, SAXException, IOException {
+        Map<String, Boolean> validation = new HashMap<>();
+        for (String example : examples.split(",")) {
+            validation.put(example, matchVariant(hugoSymbol, variant, example));
+        }
+        return new ResponseEntity<>(validation, HttpStatus.OK);
+    }
+
+    private boolean matchVariant(String hugoSymbol, String variant, String example) {
+        Gene gene = GeneUtils.getGeneByHugoSymbol(hugoSymbol);
+        boolean isMatched = false;
+        if (gene != null) {
+            AlterationBo alterationBo = ApplicationContextSingleton.getAlterationBo();
+            boolean isGeneralAlteration = AlterationUtils.isGeneralAlterations(variant);
+            Alteration oncokbVariant = new Alteration();
+            if (!isGeneralAlteration) {            // Annotate OncoKB variant
+                oncokbVariant.setGene(gene);
+                oncokbVariant.setAlteration(variant);
+                oncokbVariant.setName(variant);
+                AlterationUtils.annotateAlteration(oncokbVariant, variant);
+            }
+
+            example = example.trim();
+            Alteration exampleVariant = new Alteration();
+            exampleVariant.setGene(gene);
+            exampleVariant.setAlteration(example);
+            exampleVariant.setName(example);
+            AlterationUtils.annotateAlteration(exampleVariant, example);
+
+            LinkedHashSet<Alteration> relevantAlterations = new LinkedHashSet<>();
+            if (isGeneralAlteration) {
+                relevantAlterations = alterationBo.findRelevantAlterations(exampleVariant, true);
```

Comment:
If the general mutation is not specifically curated, it won't be returned to the relevant alteration list, which will cause the mismatch. 
For example: utils/match/variant?hugoSymbol=EGFR&variant=Gain-of-function Mutations&examples=Amplification 
It will return {"Amplification": false}, which is wrong because Amplification is curated as Gain-of-function